app/services/intercom_service.rb

```diff
@@ -129,11 +129,22 @@ def self.intercom
     @intercom ||= Intercom::Client.new(token: EnvironmentCredentials.dig(:intercom, :intercom_access_token))
   end
 
+  MAX_RETRY_COUNT = 3
+
   def self.intercom_api(collection, verb, params)
     if Rails.env.development?
       Rails.logger.debug("Calling Intercom: #{collection}.#{verb}(#{params})")
     end
-    result = intercom.send(collection).send(verb, params)
+
+    retry_counts = 0
+
+    begin
+      result = intercom.send(collection).send(verb, params)
+    rescue
+      retry_counts += 1
+      retry if retry_counts < MAX_RETRY_COUNT
+    end
```

Comment:
Is there a graceful way of surfacing this error in the instance where we've reached the `MAX_RETRY_COUNT` (datadog/sentry etc)?

Reply:
I'm waffling on this. I was considering throwing the error with an additional note in the logs about the attempted retry counts _or_ making a custom error _specific_ for this (so it's easier to alert on).

Reply:
Opted for re-raising the error with a custom message. I think with the other mention of a datadog counter, it'd be clear if it happens here